InstantPrintTool.py

```diff
@@ -38,19 +40,18 @@ def __init__(self, iface, populateCompositionFz=None):
         self.dialogui.comboBox_fileformat.addItem("BMP", self.tr("BMP Image (*.bmp);;"))
         self.dialogui.comboBox_fileformat.addItem("PNG", self.tr("PNG Image (*.png);;"))
         self.dialogui.spinBoxScale.valueChanged.connect(self.__changeScale)
-
-        self.iface.composerAdded.connect(lambda view: self.__reloadComposers())
-        self.iface.composerWillBeRemoved.connect(self.__reloadComposers)
+        self.iface.layoutDesignerOpened.connect(lambda view: self.__reloadComposers())
+        self.iface.layoutDesignerWillBeClosed.connect(self.__reloadComposers)
         self.dialogui.comboBox_composers.currentIndexChanged.connect(self.__selectComposer)
         self.exportButton.clicked.connect(self.__export)
         self.helpButton.clicked.connect(self.__help)
         self.dialogui.buttonBox.button(QDialogButtonBox.Close).clicked.connect(lambda: self.setEnabled(False))
         self.setCursor(Qt.OpenHandCursor)
 
         settings = QSettings()
-        if not settings.value("geometry") == None:
+        if settings.value("geometry") is not None:
```

Comment:
Please change geometry to "instantprint/geometry" (settings should always be scoped), and also check other uses of settings. Backport to qgis2 branch also.